src/base/bittorrent/trackerentry.h

```diff
@@ -31,31 +31,75 @@
 
 #include <libtorrent/announce_entry.hpp>
 
+#include <QString>
 #include <QtGlobal>
 
-class QString;
+class QStringList;
+template <class T> class QList;
 
 namespace BitTorrent
 {
-    class TrackerEntry
+    enum class TrackerStatus
+    {
+        NotContacted = 1,
+        Working = 2,
+        Updating = 3,
+        NotWorking = 4,
+        WorkingWarning = 5
+    };
+
+    class EndpointView
```

Comment:
You shouldn't have class definition in header.

Reply:
Sometimes it is better to inline some things like that, but I respect classic structure you have here, moved to separate files.